src/api/poll.ts

```diff
@@ -0,0 +1,85 @@
+import Bugsnag from "@bugsnag/js";
+import noop from "lodash/noop";
+
+import { PollMessage, PollValues } from "types/chat";
+
+import { getVenueRef } from "./venue";
+
+export interface CreatePollProps {
+  venueId: string;
+  poll: PollMessage;
+}
+
+export const createVenuePoll = async ({
+  venueId,
+  poll,
+}: CreatePollProps): Promise<void> =>
+  getVenueRef(venueId)
+    .collection("chats")
+    .add(poll)
+    .then(noop)
+    .catch((err) => {
+      Bugsnag.notify(err, (event) => {
+        event.addMetadata("context", {
+          location: "api/chat::createVenuePoll",
+          venueId,
+          poll,
+        });
+      });
+      // @debt rethrow error, when we can handle it to show UI error
+    });
+
+export type DeleteVenuePollProps = {
+  venueId: string;
+  pollId: string;
+};
+
+// RESTRICRED due to user rights issue
+export const deleteVenuePoll = async ({
+  venueId,
+  pollId,
+}: DeleteVenuePollProps): Promise<void> =>
+  getVenueRef(venueId)
+    .collection("chats")
+    .doc(pollId)
+    .update({ deleted: true })
+    .catch((err) => {
+      Bugsnag.notify(err, (event) => {
+        event.addMetadata("context", {
+          location: "api/chat::deleteVenuePoll",
+          venueId,
+          pollId,
+        });
+      });
+      // @debt rethrow error, when we can handle it to show UI error
+    });
```

Comment:
Please, show the delete button only to people who have both admin rights & are venue owners.

An example in `hooks/useVenueChat`, but also add the check for venue owners

Reply:
fixed in 42ff832e2b1b429a40233aa813a719c9cc6bcc96 and cec1739e63863521cb2d4921c4b3a25bcba1fda5